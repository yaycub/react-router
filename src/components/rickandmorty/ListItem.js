import React from 'react';
import PropTypes from 'prop-types';

const ListItem = ({ name, image }) => (
  <>
    <h2>{name}</h2>
    <img src={image} alt={name} />
  </>
);

ListItem.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default ListItem;
