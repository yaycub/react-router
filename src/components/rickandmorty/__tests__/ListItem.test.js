import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../ListItem';

describe('ListItem component', () => {
  it('can render a ListItem', () => {
    const wrapper = shallow(<ListItem name="a name" image="image.jpeg" />);
    expect(wrapper).toMatchSnapshot();
  });
});
