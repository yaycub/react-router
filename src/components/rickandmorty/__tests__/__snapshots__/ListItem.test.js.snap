// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component can render a ListItem 1`] = `
<Fragment>
  <h2>
    a name
  </h2>
  <img
    alt="a name"
    src="image.jpeg"
  />
</Fragment>
`;
